media/capture/content/android/java/src/org/chromium/media/ScreenCaptureService.java

```diff
@@ -384,4 +411,61 @@ private void showNotification(final CharSequence text) {
     protected PendingIntent createPendingIntent() {
         return PendingIntent.getActivity(this, 0, new Intent(this, ApplicationStatus.getLastTrackedFocusedActivity().getClass()), 0);
     }
+
+    private int getDeviceRotation() {
+        switch (mDisplay.getRotation()) {
+            case Surface.ROTATION_0:
+                return 0;
+            case Surface.ROTATION_90:
+                return 90;
+            case Surface.ROTATION_180:
+                return 180;
+            case Surface.ROTATION_270:
+                return 270;
+            default:
+                // This should not happen.
+                assert false;
+                return 0;
+        }
+    }
+
+    private @DeviceOrientation int getDeviceOrientation(int rotation) {
+        switch (rotation) {
+            case 0:
+            case 180:
+                return DeviceOrientation.PORTRAIT;
+            case 90:
+            case 270:
+                return DeviceOrientation.LANDSCAPE;
+            default:
+                // This should not happen;
+                assert false;
+                return DeviceOrientation.LANDSCAPE;
+        }
+    }
+
+    private boolean maybeDoRotation() {
+        final int rotation = getDeviceRotation();
+        final @DeviceOrientation int orientation = getDeviceOrientation(rotation);
+        if (orientation == mCurrentOrientation) {
+            return false;
+        }
+
+        Log.i(TAG, "maybeDoRotation");
+        mCurrentOrientation = orientation;
+        rotateCaptureOrientation(orientation);
+
+        final Intent result = new Intent();
+        result.setAction(ACTION_ROTATE_RESULT);
+        result.putExtra(EXTRA_RESULT_ROTATION, rotation);
+        sendBroadcast(result);
+        return true;
+    }
+
+    private void rotateCaptureOrientation(@DeviceOrientation int orientation) {
+        if ((orientation == DeviceOrientation.LANDSCAPE && mWidth < mHeight)
+                || (orientation == DeviceOrientation.PORTRAIT && mHeight < mWidth)) {
+            mWidth += mHeight - (mHeight = mWidth);
```

Comment:
Minor question.   
I just wonder if there would be a case for ``orientation == DeviceOrientation.LANDSCAPE && mWidth > mHeight``?
Wouldn't one of the option (``orientation == DeviceOrientation.LANDSCAPE`` or ``mWidth < mHeight``) enough?
Same for PORTRAIT

Reply:
orientation == DeviceOrientation.LANDSCAPE && mWidth < mHeight
The above condition is meaning that the rotation has occurred because mWidth should be larger then mHeight in the landscape mode.